### PR TITLE
support for reading vcf

### PR DESCRIPTION
**Pull request by arvkevi**
There are a couple of opinionated decisions, just something to start with and discuss.

### CODE REVIEW

File: src/lineage/snps/reader.py [hunk 1]
```diff
@@ -78,6 +79,8 @@ def __call__(self):
                 return self.read_lineage_csv(file, comments)
             elif first_line.startswith("rsid"):
                 return self.read_generic_csv(file)
+            elif "vcf" in comments.lower():
```
Comment (arvkevi) [inline src/lineage/snps/reader.py]: This was a tough one, I think it's common to look for a line staring with `"#CHROM"` to detect vcf, but there **should** be `"vcf"` in a header row per the [specs](https://samtools.github.io/hts-specs/VCFv4.2.pdf). This should work, but could be more explicit.

File: src/lineage/snps/reader.py [hunk 2]
```diff
@@ -373,3 +376,66 @@ def read_generic_csv(file):
...
+        df = pd.DataFrame(columns=["rsid", "chrom", "pos", "genotype"])
+        df = df.astype(
+            {"rsid": object, "chrom": object, "pos": np.int64, "genotype": object}
+        )
+
+        vcf_reader = vcf.Reader(open(file, "r"))
+
+        # lineage does not yet support multi-sample vcf.
+        if len(vcf_reader.samples) > 1:
+            return None, "vcf"
```
Comment (arvkevi) [inline src/lineage/snps/reader.py]: I struggled with the return here, it feels like there should be some notification the user that no file is being read, via `logging` or `warnings`?

File: src/lineage/snps/reader.py (hunk 1 above)
Comment (apriha) [reply src/lineage/snps/reader.py]: Yeah it sounds like this should work.

File: Pipfile [hunk 3]
```diff
@@ -9,6 +9,7 @@ pytest-cov = "*"
 
 [packages]
 lineage = {editable = true,path = "."}
+pyvcf = "*"
```
Comment (apriha) [inline Pipfile]: `pyvcf` isn't required here, since it'd be installed via `setup.py`

File: src/lineage/snps/reader.py (hunk 2 above)
Comment (apriha) [reply src/lineage/snps/reader.py]: Good call. I think for now a simple `print` statement should be sufficient and consistent with status printed throughout the library.

File: tests/test_individual.py [hunk 4]
```diff
@@ -24,6 +24,9 @@
 from atomicwrites import atomic_write
 import numpy as np
 import pandas as pd
+import pytest
```
Comment (arvkevi) [inline tests/test_individual.py]: This was carry over from the merge

**Comment by apriha** [comment]
Thanks @arvkevi ! This looks great. 👍

**Pull request merged**